scripts/staging/hmm/HMM Training.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import pandas as pd
+import random
+
+def add2dict(dictionary, key, value):
+    if key not in dictionary:
+        dictionary[key] = []
+    dictionary[key].append(value)
+
+def list2probabilitydict(given_list):
+    probability_dict = {}
+    given_list_length = len(given_list)
+    for item in given_list:
+        probability_dict[item] = probability_dict.get(item, 0) + 1
+    for key, value in probability_dict.items():
+        probability_dict[key] = value / given_list_length
+    return probability_dict
+
+def sample_word(dictionary):
+    key = np.random.choice(list(dictionary.keys()), 1, p = list(dictionary.values()))[0]
+    return key
+
+def generate(sentence, no_of_words):
+    sentence = sentence.split()
+    if len(sentence) == 0:
+        word0 = sample_word(initial_word)        
+        sentence.append(word0)
+
+    if len(sentence) == 1:
+        word0 = sentence[0]
+        if word0 in second_word.keys():
+            word1 = sample_word(second_word[word0])
+        else:
+            word1 = np.random.choice(list(initial_word.keys()), 1, p = list(initial_word.values()))[0]
+        sentence.append(word1)
+        no_of_words = no_of_words - 1
+    
+    if len(sentence) == 2:
```

Comment:
Make this case generic for any length of sentence i.e >=2. The idea is to predict the next word(s) based on previous k words. So the sentence could be of any length but the prediction will be based on previous k=3 words. This condition here is not accepting any input which has more than 2 words in it.